packages/babel-helper-compilation-targets/test/targets-parser.spec.js

```diff
@@ -293,6 +293,27 @@ describe("getTargets", () => {
       },
     );
 
+    (process.env.BABEL_8_BREAKING ? it.skip : it)(
+      "'browsers' option will have no effect if it is an empty array - Babel 7",
```

Comment:
I think this test has the same behavior in Babel 7 and Babel 8:
```suggestion
    it("'browsers' option will have no effect if it is an empty array",
```
also the other one.

Reply:
Yes, In your opinion, the difference of Babel 7 and Babel 8 is only `browsers` variable:

```javascript
if (browsers == null) {
      if (process.env.BABEL_8_BREAKING) {
        // In Babel 8, if no targets are passed, we use browserslist's defaults
        // and exclude IE 11.
        browsers = ["defaults, not ie 11"];
      } else {
        // If no targets are passed, we need to overwrite browserslist's defaults
        // so that we enable all transforms (acting like the now deprecated
        // preset-latest).
        browsers = [];
      }
    }
```

So if the result isn't have any effect of this, this test can also run on Babel 8.

I will update this commit later.

Thanks for comment.

Reply:
Observe the following code:
````javascript
(process.env.BABEL_8_BREAKING ? it.skip : it)(
 "'browsers' option will have no effect if it is an empty array - Babel 7", () => {
   expect(getTargets({ esmodules: "intersect", browsers: [] })).toEqual(
     getTargets({ esmodules: "intersect" }),
   );
 },
);

it("The final 'browsers' handled variable will have no effect if it is an empty array", () => {
  expect(getTargets({ esmodules: "intersect", browsers: [] })).toEqual(
    getTargets(
      { esmodules: "intersect" },
      { ignoreBrowserslistConfig: true },
    ),
  );
});
````

When `ignoreBrowserslistConfig` is set to `true`, no action is taken, but when this variable is not set and `browsers` is not set, babel7 and babel8 behave differently.

So in my opinion, the first test can not remove the conditional judgment, the second test can.